src/services/rollouts.js

```diff
@@ -0,0 +1,136 @@
+/* eslint-disable no-useless-catch */
+/* eslint-disable no-undef */
+const httpStatusCode = require('@generics/http-status')
+const responses = require('@helpers/responses')
+const common = require('@constants/common')
+const utils = require('@generics/utils')
+const rolloutQueries = require('@database/queries/rollouts')
+const resourceService = require('@services/resource')
+const resourceQueries = require('@database/queries/resources')
+module.exports = class RolloutsHelper {
+	/**
+	 * Rollout create
+	 * @method
+	 * @name create
+	 * @param {Object} req - request data.
+	 * @returns {JSON} - project id
+	 */
+	static async create(bodyData, loggedInUserId, orgId) {
+		try {
+			//validate the title length
+			const isTitleInvalid = utils.validateTitle(bodyData.title)
+			if (isTitleInvalid) {
+				return responses.failureResponse({
+					message: 'CHARACTER_LIMIT_EXCEED',
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+				})
+			}
+
+			//validation for start date and end date
+			if (bodyData.start_date && bodyData.end_date) {
+				const startDate = new Date(bodyData.start_date)
+				const endDate = new Date(bodyData.end_date)
+
+				if (startDate > endDate) {
+					return responses.failureResponse({
+						message: 'INVALID_DATE_RANGE',
+						statusCode: httpStatusCode.bad_request,
+						responseCode: 'CLIENT_ERROR',
+					})
+				}
+			}
+
+			//validate the resource
+			let resource = await resourceQueries.findOne({
+				id: bodyData.resource_id,
+				organization_id: orgId,
+				stage: common.RESOURCE_STAGE_COMPLETION,
+			})
+
+			if (!resource?.id) {
+				return responses.failureResponse({
+					message: 'RESOURCE_NOT_FOUND',
+					statusCode: httpStatusCode.bad_request,
+					responseCode: 'CLIENT_ERROR',
+				})
+			}
+
+			let rolloutData = {
+				title: bodyData.title,
+				resource_type: resource.type,
+				resource_id: resource.id,
+				status: common.ROLLOUT_STATUS_PENDING,
+				start_date: bodyData.start_date,
+				end_date: bodyData.end_date,
+				type: common.ROLLOUT_TYPE_PROGRAM,
```

Comment:
We should check the type of the resource and assign this type field. @priyanka-TL

Reply:
type is not the resource type its rollout type